lib/fs.js

```diff
@@ -454,7 +454,20 @@ function openSync(path, flags, mode) {
   return result;
 }
 
-function read(fd, buffer, offset, length, position, callback) {
+// usage:
+// fs.read(fd, [buffer[, offset[, length[, position]]], callback);
+function read(fd, ...args) {
+  let callback = args.pop();
```

Comment:
I think it might be better to avoid modifying the args array and only reading from it. From a performance perspective.

Reply:
using `args.pop` does simplify the implementation significantly,  since we know the callback is in that last slot, and it reduces the amount of type checking we would need.  But if there are going to be significant performance issues,  i'll try to re-work it

Reply:
```
callback = args[args.length - 1]
```

Reply:
@ronag it's weird, i have an email notification with a comment from you,  but i don't see it here at all